### PR TITLE
Add additional summary stat for each osp version

### PR DESCRIPTION
**Pull request by ciecierski**
User can now see both aggregated summary and separate stats for each osp version jobs.

### CODE REVIEW

**Review by odyssey4me — APPROVED** [review]
Looks sensible to me. It'd be nice to see a sample output in this PR to see how it looks.

**Comment by nathan-weinberg** [comment]
+1 to sample output please

**Comment by ciecierski** [comment]
Sample output 
screenshot-summary

**Comment by nathan-weinberg** [comment]
Can we format this so the summarys are all side-by-side?

File: jeeves/jobs.py [hunk 1]
```diff
@@ -126,9 +126,11 @@ def get_jenkins_jobs(server, job_search_fields):
...
+		returns osp version or None if no version is found
 	'''
-	version = re.search(r'\d+\.*\d*', job_name)
-	if version is None:
+	versions = re.findall(r'\b1{1}[0-7]{1}\.?\d?', job_name)
+	if not versions:
 		return None
-	return version.group()
+	versions_f = map(float, versions)
+	return max(versions_f)
```
Comment (nathan-weinberg) [inline jeeves/jobs.py:136]: Great change!

File: tests/test_jobs.py [hunk 2]
```diff
@@ -10,18 +10,20 @@ def test_get_jenkins_jobs():
...
+	assert get_osp_version('DFG-enterprise-baremetal-16.2_director-3control_2compute_externalceph-titancluster') == 16.2
+	assert get_osp_version('DFG-df-deployment-16-virthost-3cont_2comp_3ceph-ceph-ipv4-geneve-overcloud-ssl-enable') == 16
+	assert get_osp_version('DFG-hardware_provisioning-rqci-16_director-rhel-8.1-spineleaf-provision-network-20191205-2116') == 16
+	assert get_osp_version('DFG-network-networking-ovn-16_director-rhel-virthost-3cont_2comp-ipv6-geneve') == 16
+	assert get_osp_version('DFG-osasinfra-shiftstack_on_vms-16_director-rhel-virthost-3cont_1comp-ipv4-geneve-hybrid_flat_4.3_3master_3worker-ovn') == 16
+	assert get_osp_version('DFG-pidone-sanity-16_director-rhel-virthost-3cont_2comp-ipv4-geneve-tobiko_faults-sanity') == 16
+	assert get_osp_version('DFG-security-keystone-16_director-rhel-virthost-1cont_1comp-ipv4-geneve-lvm-containers') == 16
+	assert get_osp_version('DFG-upgrades-updates-16-from-passed_phase1-HA-ipv4') == 16
+	assert get_osp_version('DFG-upgrades-updates-from-13-to-16.2-passed_phase1-HA-ipv4') == 16.2
+	assert get_osp_version('DFG-enterprise-baremetal-13_director-3control_2compute-titancluster') == 13
```
Comment (nathan-weinberg) [inline tests/test_jobs.py:28]: And thank you for also updating the test!

File: jeeves/jobs.py [hunk 3]
```diff
@@ -126,9 +126,11 @@ def get_jenkins_jobs(server, job_search_fields):
 
 def get_osp_version(job_name):
 	''' gets osp version from job name via regex
-		returns None if no version is found
+		if multiple versions detected, the highest number is considered osp version and
+		returns osp version or None if no version is found
 	'''
-	version = re.search(r'\d+\.*\d*', job_name)
-	if version is None:
+	versions = re.findall(r'\b1{1}[0-7]{1}\.?\d?', job_name)
```
Comment (itzikb) [inline jeeves/jobs.py:132]: This won't work with a job name such as 'somejob-osp13'

**Review by itzikb — CHANGES_REQUESTED** [review]

File: jeeves/jobs.py (hunk 3 above)
Comment (ciecierski) [reply jeeves/jobs.py:132]: I will try to fix the regex

**Comment by odyssey4me** [comment]
I've been thinking about this quite a bit, and the more I look at the report for a range of versions the more I think that it would be best to rather split out each version into its own summary table - rather than combine everything into a single table like it does now. @nathan-weinberg What do you think?

**Comment by ciecierski** [comment]
> I've been thinking about this quite a bit, and the more I look at the report for a range of versions the more I think that it would be best to rather split out each version into its own summary table - rather than combine everything into a single table like it does now. @nathan-weinberg What do you think?

That's interesting idea I like it too. When the table is uploaded to confluence you can use confluence sorting, however it still hard to track easily what is going on.

**Comment by nathan-weinberg** [comment]
> > I've been thinking about this quite a bit, and the more I look at the report for a range of versions the more I think that it would be best to rather split out each version into its own summary table - rather than combine everything into a single table like it does now. @nathan-weinberg What do you think?
> 
> That's interesting idea I like it too. When the table is uploaded to confluence you can use confluence sorting, however it still hard to track easily what is going on.

It is an interesting idea - I think the best path forward would be a POC so we can see it and opening up the discussion to all of the users. I'm not sure if we want to make that part of this PR though - it may work better as a followup.

**Comment by ciecierski** [comment]
After fix:
table-after

**Review by itzikb — COMMENTED** [review]
Thanks it works for me.
If the headers of the summary can be aligned it will be great
image

**Review by EliadCohen — APPROVED** [review]
I love it, looks good to me

**Review by nathan-weinberg — CHANGES_REQUESTED** [review]
Tested with Enterprise DFG jobs - a few requests:

1. Capitalize all job statuses (e.g. "success" => "SUCCESS") and include "= <percent>" - basically, mimic the style of the overall summary (apart from blocker/blocker coverage, we don't need that on a per-version basis i dont think)
2. Add "NO_KNOWN_BUILDS" to version summaries
3. "13" is showing as "13.0" now - let's put this back to what is was, only 16.1 and 16.2 should be showing decimal values

Everything else is looking great, thank you for your continued work on this @ciecierski

**Comment by ciecierski** [comment]
> Tested with Enterprise DFG jobs - a few requests:
> 
> 1. Capitalize all job statuses (e.g. "success" => "SUCCESS") and include "= " - basically, mimic the style of the overall summary (apart from blocker/blocker coverage, we don't need that on a per-version basis i dont think)
> 2. Add "NO_KNOWN_BUILDS" to version summaries
> 3. "13" is showing as "13.0" now - let's put this back to what is was, only 16.1 and 16.2 should be showing decimal values
> 
> Everything else is looking great, thank you for your continued work on this @ciecierski

Please take a look at the latest code changes. It addresses all your requests. Thank you

**Comment by ciecierski** [comment]
> Thanks it works for me.
> If the headers of the summary can be aligned it will be great
> image

Should be fine now

**Review by nathan-weinberg — CHANGES_REQUESTED** [review]
Looking good - however OSP13 is still showing at "13.0"

File: jeeves/report.py [hunk 4]
```diff
@@ -12,6 +12,36 @@
...
+		summary['total_missing'] = False
+
+	# include error report if needed
+	if num_error > 0:
+		summary['total_error'] = "ERROR:  {}/{} = {}%".format(num_error, num_jobs, percent(num_error, num_jobs))
+	else:
+		summary['total_error'] = False
+	return summary
+
+
```
Comment (nathan-weinberg) [inline jeeves/report.py:44]: Move to jeeves.common - maybe we can call this `generate_summary`?

Comment (ciecierski) [reply jeeves/report.py]: Done

**Comment by nathan-weinberg** [comment]
Thanks - please address my comments about 13 and I think we should be gtg from my end

File: templates/report_template.html [hunk 5]
```diff
@@ -22,23 +22,35 @@
...
 							{% endif %}
 							<li>{{summary.total_bugs}}</li>
 							<li>{{summary.total_tickets}}</li>
-							<li>{{summary.total_coverage}}</li>
+							<li>Total {{summary.total_coverage}}</li>
 						</ul>
 					</td>
+					{% for sorted_version in stats_per_version.keys()|sort|reverse %}
+					<td>
+						<p><b>Summary for {{ (sorted_version|string).rstrip('.0') }}</b></p>
```
Comment (ciecierski) [inline templates/report_template.html:44]: @nathan-weinberg 13 is addressed here already
after-fix

**Comment by ciecierski** [comment]
> Thanks - please address my comments about 13 and I think we should be gtg from my end
It is already adressed. 
after-fix


> Thanks - please address my comments about 13 and I think we should be gtg from my end

done

File: templates/report_template.html (hunk 5 above)
Comment (nathan-weinberg) [reply templates/report_template.html:44]: I'm talking about in the report table

**Comment by nathan-weinberg** [comment]
> > Thanks - please address my comments about 13 and I think we should be gtg from my end
> > It is already adressed.
> > after-fix
> 
> > Thanks - please address my comments about 13 and I think we should be gtg from my end
> 
> done

Report table still shows "13.0"

File: templates/report_template.html (hunk 5 above)
Comment (ciecierski) [reply templates/report_template.html]: Sorry, my misunderstanding. It is fixed now both in the table and summary.

**Review by nathan-weinberg — APPROVED** [review]

**Pull request merged**